metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/usage.py

```diff
@@ -160,113 +345,151 @@ def __init__(self, config: BigQueryV2Config, report: BigQueryV2Report):
         self.config: BigQueryV2Config = config
         self.report: BigQueryV2Report = report
 
-    def add_config_to_report(self):
-        self.report.query_log_delay = self.config.usage.query_log_delay
-
     def _is_table_allowed(self, table_ref: Optional[BigQueryTableRef]) -> bool:
         return (
             table_ref is not None
             and self.config.dataset_pattern.allowed(table_ref.table_identifier.dataset)
             and self.config.table_pattern.allowed(table_ref.table_identifier.table)
         )
 
-    def generate_usage_for_project(
-        self, project_id: str, tables: Dict[str, List[str]]
+    def run(
+        self, projects: Iterable[str], table_refs: Collection[str]
     ) -> Iterable[MetadataWorkUnit]:
-        aggregated_info: Dict[
-            datetime, Dict[BigQueryTableRef, AggregatedDataset]
-        ] = collections.defaultdict(dict)
+        events = itertools.chain.from_iterable(
+            self._get_usage_events(project) for project in projects
+        )
+        yield from self._run(events, table_refs)
 
-        parsed_events: Iterable[Union[ReadEvent, QueryEvent]]
-        with PerfTimer() as timer:
+    def _run(
+        self, events: Iterable[AuditEvent], table_refs: Collection[str]
+    ) -> Iterable[MetadataWorkUnit]:
+        try:
+            with BigQueryUsageState(self.config) as usage_state:
+                self._ingest_events(events, table_refs, usage_state)
+
+                if self.config.usage.include_operational_stats:
+                    yield from self._generate_operational_workunits(usage_state)
+
+                yield from self._generate_usage_workunits(usage_state)
+        except Exception:
+            logger.error("Error processing usage", exc_info=True)
+
+    def _ingest_events(
+        self,
+        events: Iterable[AuditEvent],
+        table_refs: Collection[str],
+        usage_state: BigQueryUsageState,
+    ) -> None:
+        """Read log and store events in usage_state."""
+        num_aggregated = 0
+        for audit_event in events:
             try:
-                bigquery_log_entries = self._get_parsed_bigquery_log_events(project_id)
-                if self.config.use_exported_bigquery_audit_metadata:
-                    parsed_events = self._parse_exported_bigquery_audit_metadata(
-                        bigquery_log_entries
-                    )
-                else:
-                    parsed_events = self._parse_bigquery_log_entries(
-                        bigquery_log_entries
-                    )
+                num_aggregated += self._store_usage_event(
+                    audit_event, usage_state, table_refs
+                )
+            except Exception:
+                logger.warning(
```

Comment:
Now handled by `report.num_usage_failures` dict

Reply:
Swapping to `report_warning`, didn't realize how big a deal a failure is, oops. I want to keep in the report just because logs get truncated and are unreliable